code/modules/recycling/disposal-construction.dm

```diff
@@ -96,230 +93,230 @@
-			if(10)
-				switch(subtype)
-					if(0)
-						return /obj/structure/disposalpipe/sortjunction/flipped
-					if(1)
-						return /obj/structure/disposalpipe/sortjunction/wildcard/flipped
-					if(2)
-						return /obj/structure/disposalpipe/sortjunction/untagged/flipped
+/obj/structure/disposalconstruct/proc/dpipetype()
+	switch(pipe_type)
+		if(PIPE_TYPE_STRAIGHT, PIPE_TYPE_BENT)
+			return /obj/structure/disposalpipe/segment
+		if(PIPE_TYPE_JUNC, PIPE_TYPE_JUNC_FLIP, PIPE_TYPE_JUNC_Y)
+			return /obj/structure/disposalpipe/junction
+		if(PIPE_TYPE_TRUNK)
+			return /obj/structure/disposalpipe/trunk
+		if(PIPE_TYPE_BIN)
+			return /obj/machinery/disposal
+		if(PIPE_TYPE_OUTLET)
+			return /obj/structure/disposaloutlet
+		if(PIPE_TYPE_INTAKE)
+			return /obj/machinery/disposal/deliveryChute
+		if(PIPE_TYPE_JUNC_SORT)
+			switch(sort_type)
+				if(SORT_TYPE_NORMAL)
+					return /obj/structure/disposalpipe/sortjunction
+				if(SORT_TYPE_WILDCARD)
+					return /obj/structure/disposalpipe/sortjunction/wildcard
+				if(SORT_TYPE_UNTAGGED)
+					return /obj/structure/disposalpipe/sortjunction/untagged
+		if(PIPE_TYPE_JUNC_SORT_FLIP)
+			switch(sort_type)
+				if(SORT_TYPE_NORMAL)
+					return /obj/structure/disposalpipe/sortjunction/flipped
+				if(SORT_TYPE_WILDCARD)
+					return /obj/structure/disposalpipe/sortjunction/wildcard/flipped
+				if(SORT_TYPE_UNTAGGED)
+					return /obj/structure/disposalpipe/sortjunction/untagged/flipped
 ///// Z-Level stuff
-			if(11)
-				return /obj/structure/disposalpipe/up
-			if(12)
-				return /obj/structure/disposalpipe/down
+		if(PIPE_TYPE_UP)
+			return /obj/structure/disposalpipe/up
+		if(PIPE_TYPE_DOWN)
+			return /obj/structure/disposalpipe/down
 ///// Z-Level stuff
-			if(13)
-				return /obj/structure/disposalpipe/tagger
-			if(14)
-				return /obj/structure/disposalpipe/tagger/partial
-		return
+		if(PIPE_TYPE_TAGGER)
+			return /obj/structure/disposalpipe/tagger
+		if(PIPE_TYPE_TAGGER_PART)
+			return /obj/structure/disposalpipe/tagger/partial
+	return
 
 
 
 	// attackby item
 	// wrench: (un)anchor
 	// weldingtool: convert to real pipe
 
-	attackby(var/obj/item/I, var/mob/user)
-		var/nicetype = "pipe"
-		var/ispipe = 0 // Indicates if we should change the level of this pipe
-		src.add_fingerprint(user)
-		switch(ptype)
-			if(6)
-				nicetype = "disposal bin"
-			if(7)
-				nicetype = "disposal outlet"
-			if(8)
-				nicetype = "delivery chute"
-			if(9, 10)
-				switch(subtype)
-					if(0)
-						nicetype = "sorting pipe"
-					if(1)
-						nicetype = "wildcard sorting pipe"
-					if(2)
-						nicetype = "untagged sorting pipe"
-				ispipe = 1
-			if(13)
-				nicetype = "tagging pipe"
-				ispipe = 1
-			if(14)
-				nicetype = "partial tagging pipe"
-				ispipe = 1
-			else
-				nicetype = "pipe"
-				ispipe = 1
+/obj/structure/disposalconstruct/attackby(var/obj/item/I, var/mob/user)
+	var/nicetype = "pipe"
+	var/ispipe = 0 // Indicates if we should change the level of this pipe
```

Comment:
Please rename variable in accordance with snake case style and use TRUE/FALSE values because it is using as boolean.